### ISSUE TITLE
Assert on process shutdown while tracing C implementation of EventPipe

### ISSUE THREAD

**Issue by josalem** [dotnet/runtime#46159]
A checked build of the C implementation of EventPipe triggers an assert during process shutdown if the app has a tracing session open.

```
Assert failure(PID 21410 [0x000053a2], Thread: 23643989 [0x168c755]): thread->session_state [i] == 0
    File: /Users/josalem/git/alts/runtime/src/native/eventpipe/ep-thread.c Line: 64
    Image: /Users/josalem/git/EventPipeLab/scalability/corescaletest/c-checked/corescaletest
```

This is a valid scenario that works under the C++ implementation.

Belongs to #46079 

CC - @lateralusX

**Comment by lateralusX** [comment]
Believe this is a different way that the issue manifests under checked build but the underlying issue is most likely the same as for https://github.com/dotnet/runtime/issues/46158.

**Comment by lateralusX** [comment]
Run session for 60 seconds and then shutdown runtime, no assert on shutdown after fixes.

**Comment by lateralusX** [comment]
https://github.com/dotnet/runtime/pull/46193

**Comment by josalem** [comment]
closed via #46193
